lib/DXIL/DxilNodeProps.cpp

```diff
@@ -81,6 +90,16 @@ bool NodeFlags::GetTrackRWInputSharing() const {
           (uint32_t)DXIL::NodeIOFlags::TrackRWInputSharing) != 0;
 }
 
+void NodeFlags::SetGloballyCoherent() {
+  m_Flags = (DXIL::NodeIOFlags)((uint32_t)m_Flags |
```

Comment:
nit: I know it is more verbose, but it is generally preferable to use `static_cast<>` over the C-style cast. The C++ casts have more clearly defined behavior so they are less prone to misuse or unexpected behavior.

Reply:
I was being consistent with the existing code here.  The use of `static_cast<>` for this kind of case is so ugly.  The noise drowns out the signal, making the code harder to read.  I think I'd rather implement the changes to the enum and/or global operators discussed in the other comment to avoid the ugly casting in the first place.

Reply:
That's fair, but "ugly" code that is correct and explicit should always be preferred over brief code that has less clear meaning. Especially since the ugliness of code is extremely subjective, but the explicitness or implicitness isn't. I generally think c-style casts are ugly, which I'm guessing is a point you and I disagree on.

Reply:
Oh, I agree generally, but when it comes to code using `&`/`|` on flag values, using static_cast usually ends up making it harder to read (and notice faults in) the code.  Is there a practical advantage to it in this use case (and of using enum class instead of enum in this NodeFlags class for the flags)?

Reply:
I'll revisit the flags issue in separate PR, since this one is just adding code with exactly the same pattern, and I'd rather not mix the concern of the best way to solve that with this PR.